### DISCUSSION TITLE
ANT vs Gradle vs Maven as the main build script?

### DISCUSSION THREAD

**Discussion by radek2s** [SCADA-LTS/Scada-LTS#1492] [Ideas]
Scada-LTS project from the beginning is using ANT library to manage dependencies and build sequence. But in 2.5.0 version release was introduced "beta" gradle configuration to build the project and sub-project like "scadalts-ui". I think that is the best place to share experience which one should be the main build configuration.

Short description of each of them:
**Apache ANT**
Key feature is to drive processes described in build files as targets. It is written in Java and is extremely flexible which means that do not impose coding conventions or directory layouts to the project. Ant supplies a number of build-in tasks allowing to compile, assemble, test and run Java applications. [~Apache Ant Welcome page](https://ant.apache.org/)
**Apache Maven**
Maven is a Java tool for building and managing Java-base projects. Objective of Maven is to make the build process easy and provide a uniform build system with project information. It encourage better development practices. As an advantage of that solution Maven can provide dependencies list used by specific project and many useful project information. [~Apache Maven - What is maven?](https://maven.apache.org/what-is-maven.html)
**Gradle**
Gradle is also a build tool but not only for Java. It can be used for native C/C++ development or with Node.JS integration. Like Maven, Gradle provide convention over configuration approach but Gradle is more advanced tool that is designed for empowered and responsible user. Gradle also provide a dependency management. 

So what are your favorites? Do you see pros and cons of each approach?

**Comment by grzesiekb** [comment]
Ultimately gradle has the most features and it combines the advantages of both ant and maven. But traditionally builds with ant because we don't use library repositories, I guess.
